### ISSUE TITLE
Implement `spago script`

### ISSUE THREAD

**Issue by f-f** [purescript/spago#1000]
Still needs to be ported from the Haskell implementation, see [here](https://github.com/purescript/spago-legacy/blob/11e68ece50ede286660452ef370baa3ab5c33c02/src/Spago/Build.hs#L269)

**Comment by JordanMartinez** [comment]
This may require changes in `node-execa` first as I don't think it'll handle multi-line shebangs properly right now (or this might only be the case on Windows). See https://github.com/JordanMartinez/purescript-node-execa/blob/master/src/Node/Library/Execa/CrossSpawn.purs#L181-L188

**Comment by f-f** [comment]
This should now be unblocked?

**Comment by JordanMartinez** [comment]
Yes and no. For Linux/Mac, this is unblocked. For Windows, this is unblocked but may not work correctly in all cases. In `node-execa` (as noted above), we only read the first 150 bytes of a file. If there are multi-line shebangs that make up the entire command, we'll drop some of that on Windows.

**Comment by f-f** [comment]
Do we know why that 150? Is that just a number that is deemed "good enough" to read all the shebang lines?

**Comment by JordanMartinez** [comment]
> Is that just a number that is deemed "good enough" to read all the shebang lines?

AFAICT, yes. That's what was in the original code. I can't recall whether there was a comment explaining why.

**Comment by f-f** [comment]
150 chars should be quite enough for the majority of cases, we can try shipping this and come back to bump it if we ever figure out it's not enough

**Comment by JordanMartinez** [comment]
It's your call.

**Comment by f-f** [comment]
Let's

**Comment by fsoikin** [comment]
A question/proposal: instead of (or in addition to) specifying compiler version and dependencies on the command line, how about  supporting then via pragmas in the file itself? [Haskell Stack does that](https://docs.haskellstack.org/en/stable/topics/scripts) to an extent, but I don't like the command-line syntax, I think [F# does it much better](https://learn.microsoft.com/en-us/dotnet/fsharp/tools/fsharp-interactive/#scripting-with-f). 

And speaking of F#: it has another pragma `#load` that allows to have multi-module scripts, and could be trivially implemented in Spago - just copy all those `#load`ed files to the same temp dir.

**Comment by f-f** [comment]
I don't have a strong opinion on this, except for the fact that whatever we add in `.purs` files needs to be supported by the parser. E.g. when we introduced `spago script` for the Haskell implementation we [had to patch the parser](https://github.com/purescript/purescript/pull/4214)

**Comment by fsoikin** [comment]
Yes, of course, the pragmas would look like comments to the compiler. 

The Haskell patch was for shebang lines, not the pragmas. Pragmas in Haskell files are also just comments, no patch would be needed for them.

And while I wasn't actually proposing to support shebangs as well, it can also be easily done: as long as we're copying the files to a different location anyway, we can just cut off the shebang lines along the way.

**Comment by pete-murphy** [comment]
Hi, I've just made this issue https://github.com/purescript/spago/issues/1403, which could be resolved by removing `spago script` from the README, but then I saw this issue about reinstating it. Is there still interest in having this command? It appears to be unblocked if I'm understanding the conversation here.
